src/fmu/dataio/dataio.py

```diff
@@ -942,11 +941,17 @@ def _update_settings(self, newsettings: dict) -> None:
         legals = {key: val for key, val in annots.items() if not key.startswith("_")}
 
         for setting, value in newsettings.items():
-            if _validate_variable(setting, value, legals):
-                setattr(self, setting, value)
-                if setting == "verbosity":
-                    logger.setLevel(level=self.verbosity)
-                logger.info("New setting OK for %s", setting)
+            if setting == "restart_from":
+                warn(
+                    "The 'restart_from' has been move from case to iteration. "
```

Comment:
I don't think this makes sense to the end user. Suggest something like

"The 'restart_from' argument is deprecated and will be removed in a future version. Please refer to the fmu-dataio documentation for information on how to record information about restart source."

The main message we want to give to the user is that he/she should stop using this argument, they would not know what "moved from case to iteration" means I think.

Reply:
Agree and implemented